pkg/manager/app/handler.go

```diff
@@ -44,38 +45,33 @@ func (p *Server) DescribeApps(ctx context.Context, req *pb.DescribeAppsRequest)
 		Select(models.AppColumns...).
 		From(models.AppTableName).
 		Offset(offset).
-		Limit(limit)
-	// TODO: filter condition
-	if len(req.GetName()) > 0 {
-		query = query.Where(db.Eq("name", req.GetName()))
-	}
-	if len(req.GetRepoId()) > 0 {
-		query = query.Where(db.Eq("repo_id", req.GetRepoId()))
-	}
-	if len(req.GetAppId()) > 0 {
-		query = query.Where(db.Eq("app_id", req.GetAppId()))
-	}
-	if len(req.GetStatus()) > 0 {
-		query = query.Where(db.Eq("status", req.GetStatus()))
-	}
-
-	count, err := query.Load(&apps)
+		Limit(limit).
+		Where(manager.BuildFilterConditions(req, models.AppColumns...))
+	_, err := query.Load(&apps)
 	if err != nil {
 		// TODO: err_code should be implementation
 		return nil, status.Errorf(codes.Internal, "DescribeApps: %+v", err)
 	}
+	count, err := query.Count()
```

Comment:
Why use another query to get count?

Reply:
The previous code can only get the value of the current row, and it is wrong.

Reply:
count = len(apps)?

Reply:
OK, I know, count means total count, without consideration of limit and offset.